### PR TITLE
Fix for Websocket behind Proxy Issue

### PR DESCRIPTION
**Pull request by ProfessorMc**
Fix for #194 

Per google documentation suggestion, updated the websocket library.  Now setting http_proxy and https_proxy will be honored by the driver and passed correctly to AMQP library.

### CODE REVIEW

**Comment by ProfessorMc** [comment]
Hold off review please, need to make 1 additional change to error handle on disconnect.

File: namespace.go [hunk 1]
```diff
@@ -190,13 +191,17 @@ func (ns *Namespace) newClient() (*amqp.Client, error) {
 
 	if ns.useWebSocket {
 		wssHost := ns.getWSSHostURI() + "$servicebus/websocket"
-		wssConn, err := websocket.Dial(wssHost, "amqp", "http://localhost/")
+		ctx, cancelfunc := context.WithTimeout(context.Background(), 10*time.Second)
```
Comment (jhendrixMSFT) [inline namespace.go:194]: Can you please flow the context from the callers instead of hard-coding this.

File: receiver.go [hunk 2]
```diff
@@ -47,6 +47,7 @@ type (
 		useSessions        bool
 		sessionID          *string
 		lastError          error
+		lastErrorMu        sync.RWMutex
```
Comment (jhendrixMSFT) [inline receiver.go:50]: Is there a reason we can't use `clientMu` when reading/writing last error?

Also curious if you observed actual race here or this is a proactive change.

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
Flow context.

File: receiver.go (hunk 2 above)
Comment (ProfessorMc) [reply receiver.go:50]: This was changed by one of my dev team.  You can pass in the race flag and observe it. @alexpashkov can you comment on the mutex question?

File: namespace.go (hunk 1 above)
Comment (ProfessorMc) [reply namespace.go:219]: @jhendrixMSFT do you mean pass the context in as part of the WithWebSocket option, or do you mean changing the method call to accept ctx?

Comment (jhendrixMSFT) [reply namespace.go:219]: update the containing method `newClient()` to take a context parameter (its callers already take a context).

Comment (ProfessorMc) [reply namespace.go:219]: @jhendrixMSFT Okay, working on it now.  I will add another option to set the connection timeout as well, I don't really care for the hard-coded 10 second either.

Comment (jhendrixMSFT) [reply namespace.go:219]: I don't think you need to set any timeout, just accept whatever context the caller passes us (the caller controls the lifetime).

File: receiver.go (hunk 2 above)
Comment (alexpashkov) [reply receiver.go:50]: > Is there a reason we can't use `clientMu` when reading/writing last error?
> 
> Also curious if you observed actual race here or this is a proactive change.

I've checked the code and seems like `clientMu` is used for absolutely different things, so `lastErrorMu` is reasonable. There is a race, you can observe it with a `-race` flag

Comment (jhendrixMSFT) [reply receiver.go:50]: Thanks everybody for the comments on this.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**